GVFS/GVFS.Platform.Windows/WindowsFileSystemVirtualizer.cs

```diff
@@ -504,46 +507,35 @@ private HResult EndDirectoryEnumerationHandler(Guid enumerationId)
                     activeEnumeration.TrySaveFilterString(filterFileName);
                 }
 
-                bool entryAdded = false;
-
                 HResult result = HResult.Ok;
+                bool entryAdded = false;
                 while (activeEnumeration.IsCurrentValid)
                 {
                     ProjectedFileInfo fileInfo = activeEnumeration.Current;
                     FileProperties properties = this.FileSystemCallbacks.GetLogsHeadFileProperties();
 
-                    result = results.Add(
+                    bool addResult = results.Add(
                         fileName: fileInfo.Name,
-                        fileSize: (ulong)(fileInfo.IsFolder ? 0 : fileInfo.Size),
+                        fileSize: fileInfo.IsFolder ? 0 : fileInfo.Size,
                         isDirectory: fileInfo.IsFolder,
-                        fileAttributes: fileInfo.IsFolder ? (uint)NativeMethods.FileAttributes.FILE_ATTRIBUTE_DIRECTORY : (uint)NativeMethods.FileAttributes.FILE_ATTRIBUTE_ARCHIVE,
+                        fileAttributes: fileInfo.IsFolder ? FileAttributes.Directory : FileAttributes.Archive,
                         creationTime: properties.CreationTimeUTC,
                         lastAccessTime: properties.LastAccessTimeUTC,
                         lastWriteTime: properties.LastWriteTimeUTC,
                         changeTime: properties.LastWriteTimeUTC);
 
-                    if (result == HResult.Ok)
+                    if (addResult == true)
                     {
                         entryAdded = true;
                         activeEnumeration.MoveNext();
                     }
-                    else if (result == HResult.InsufficientBuffer)
+                    else

```

Comment:
@wilbaker @cgallred was this the one we discussed.  I.e. it only returns false on InsufficentBuffer?

Reply:
We should be able to simplify this to:

```
                if (results.Add(...))
                {
                    entryAdded = true;
                    enumeration.MoveNext();
                }
                else
                {
                    if (entryAdded)
                    {
                        hr = HResult.Ok;
                    }
                    else
                    {
                        hr = HResult.InsufficientBuffer;
                    }

                    break;
                }
```

Reply:
More details on the return value for `Add`:

```
    /// <returns>
    ///     <para>
    ///     <c>true</c> if the entry was successfully added to the enumeration buffer, <c>false</c> otherwise.
    ///     </para>
    ///     <para>
    ///     If this method returns <c>false</c>, the provider returns <see cref="HResult::Ok"/> and waits for
    ///     the next <c>GetDirectoryEnumerationCallback</c>.  Then it resumes filling the enumeration with
    ///     the entry it was trying to add when it got <c>false</c>. 
    ///     </para>
    ///     <para>
    ///     If the function returns <c>false</c> for the first file or directory in the enumeration, the
    ///     provider returns <see cref="HResult::InsufficientBuffer"/> from the <c>GetDirectoryEnumerationCallback</c>
    ///     method.
    ///     </para>
    /// </returns>
```